src/app/shared/cookies/browser-klaro.service.ts

```diff
@@ -68,14 +80,22 @@ export class BrowserKlaroService extends KlaroService {
       this.klaroConfig.translations.en.consentNotice.description = 'cookies.consent.content-notice.description.no-privacy';
     }
 
+    const configurationToHide$: Observable<Pick<typeof klaroConfiguration, 'name'>[]> =
+      this.configService.findByPropertyName(this.GOOGLE_ANALYTICS_KEY)
+        .pipe(
+          getFirstCompletedRemoteData(),
+          map(remoteData => this.mapInvalidConfiguration(remoteData, this.GOOGLE_ANALYTICS_SERVICE_NAME)),
+          take(1)
+        );
```

Comment:
Hi @ybnd, thank you for the review.

> - I'm not sure what your reason is for the Pick<...> type, as far as I can tell we can achieve the same with string[]

The reason is to clearify that we are expecting the same property type as the one named `name` in the `klaroConfiguration` object, in this way we can easily compare them using the `===` (triple equal) with the property of the object `BrowserKlaroService.filterConfigServices`.

> - Please pull out this entire pipe into a separate method, inlining the two `mapInvalidConfiguration` and `isEmptyOrInvalid` into the map operator.
IMO this would make it more clear and testable.

Do you mean the three operators of the pipe for _entire pipe_?

What do you mean for: _inlining the two `mapInvalidConfiguration` and `isEmptyOrInvalid` into the map operator_?
- Replace the `mapInvalidConfiguration` inside the `map` operator with the body of its definition, where we have the `isEmptyOrInvalid` check?

Reply:
But `klaroConfiguration` doesn't have a `name` key, and it's typed as `any` anyway? 

Right now that type only enforces the `name` property itself, but doesn't add any type safety -- you can just pass it a number for instance.
I agree that it would be nice to "synchronize" types between the configuration object and the names we use to hide stuff from it, but I don't think the current implementation does that. Instead, it makes the logic harder to understand.

So I suggest simplifying this into something like
```typescript
const servicesToHide$: Observable<string[]> = this.configService.findByPropertyName(this.GOOGLE_ANALYTICS_KEY).pipe(
  getFirstCompletedRemoteData(),
  // map(remoteData => this.mapInvalidConfiguration(remoteData, this.GOOGLE_ANALYTICS_SERVICE_NAME)),
  map(remoteData => {
    if (!remoteData.hasSucceeded || !remoteData.payload || isEmpty(remoteData.payload.values)) {
      return [this.GOOGLE_ANALYTICS_SERVICE_NAME];
    } else {
      return [];
    }
  }),
);
```
* Don't encapsulate the `name` in an object
* Keeps everything close together so its easier to follow at a glance

(yes, that's what I meant by inlining those methods)

Reply:
Ok, understood.
Thank you!